content/geoip2/web-services/using-the-javascript-client.mdx

```diff
@@ -0,0 +1,241 @@
+[register domains directly](https://www.maxmind.com/en/javascript_domains).
+
+### 2. Add a JavaScript snippet to your page
+
+In order to use this service, the following JavaScript must be included in your
+page.
+
+**Do not download this JavaScript file and load it from your server(s)!**
+
+```html
+<script src="//geoip-js.com/js/apis/geoip2/v2.1/geoip2.js" type="text/javascript"></script>
+```
+
+### 3. Call an API method and provide callbacks
+
+The JavaScript client provides 3 public methods:
+
+| Option                                         | Description                                                                                                                     |
+| ---------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| `geoip2.country(onSuccess, onError, options)`  | Calls the GeoIP2 Precision: Country endpoint using the routable IP address associated with the machine on which it is running.  |
+| `geoip2.city(onSuccess, onError, options)`     | Calls the GeoIP2 Precision: City endpoint using the routable IP address associated with the machine on which it is running.     |
+| `geoip2.insights(onSuccess, onError, options)` | Calls the GeoIP2 Precision: Insights endpoint using the routable IP address associated with the machine on which it is running. |
+
+All of the functions take the same 3 arguments:
+
+* If successful, this function calls the `onSuccess`. The first parameter
+
+passed to the callback contains an object matching the output of one of
+MaxMind’s [Precision web service API
+responses](https://dev.maxmind.com/geoip/geoip2/web-services/).
+
+In addition to the attributes listed in the web services API docs, the object
+also has a `most_specific_subdivision` property that provides access to the
+most specific subdivision object available. On browsers other than Internet
+Explorer 8, this is implemented as a non-enumerable property using
+`defineProperty`.
+
+* If there is an error, the function calls the `onError` with the error object
+
+as a parameter.
+
+* The optional `options` parameter is an object containing flags for the
+
+service. This parameter is reserved for future use. There are no options at
+this time.
+
+### Example: Display the user's city name in a page
+
+In this example, we are displaying the user's city name in a page.  We have 2
+files: page.html and demo.js.
+
+```html
+<!-- page.html -->
+<!doctype html>
+<html>
+  <head>
+    <title>Which city am I in?</title>
+    <script src="//geoip-js.com/js/apis/geoip2/v2.1/geoip2.js"></script>
+    <!-- make sure this is after the geoip2.js file -->
+    <script src="/demo.js"></script>
+  </head>
+
+  <body>
+    <p>
+       Where am I?
+    </p>
+
+    <p>
+      You are in <span id="city"></span>.
+    </p>
+  </body>
+</html>
+```
+
+```js
+// demo.js
+var fillInPage = (function() {
```

Comment:
Should we use two spaces to condense the indention a bit? We are using two spaces in the html block above this as well as in the json block below this.